apis/externalsecrets/v1beta1/secretstore_onepassword_sdk_types.go

```diff
@@ -0,0 +1,47 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	esmeta "github.com/external-secrets/external-secrets/apis/meta/v1"
+)
+
+// OnePasswordSDKAuth contains a secretRef for the service account token.
+type OnePasswordSDKAuth struct {
+	// ServiceAccountSecretRef points to the secret containing the token to access 1Password vault.
+	ServiceAccountSecretRef esmeta.SecretKeySelector `json:"serviceAccountSecretRef"`
+}
+
+// IntegrationInfo specifies the name and version of the integration built using the 1Password Go SDK.
+type IntegrationInfo struct {
+	// Name defaults to DefaultIntegrationName.
+	// +kubebuilder:default="DefaultIntegrationName"
+	Name string `json:"name,omitempty"`
+	// Version defaults to DefaultIntegrationVersion.
+	// +kubebuilder:default="DefaultIntegrationVersion"
+	Version string `json:"version,omitempty"`
+}
+
+// OnePasswordSDKProvider configures a store to sync secrets using the 1Password sdk.
+type OnePasswordSDKProvider struct {
+	// Vault defines the vault's name to access. Do NOT add op:// prefix. This will be done automatically.
+	Vault string `json:"vault"`
+	// IntegrationInfo specifies the name and version of the integration built using the 1Password Go SDK.
+	// If you don't know which name and version to use, use `DefaultIntegrationName` and `DefaultIntegrationVersion`, respectively.
+	// +optional
+	IntegrationInfo IntegrationInfo `json:"integrationInfo,omitempty"`
```

Comment:
If this is optional, I find it better for it to also be a `nil` pointer. Otherwise tools like argocd will try to sync this manifest without it, ESO will patch it and add a default IntegrationInfo causing out of sync loops (for some reason still not clear to me)

Reply:
I was thinking about that indeed. But since both of the fields are technically required but have default values, I thought this is good for that?

Reply:
I'll try later today (when I'm really up) to see what happens when I `kubectl apply` it. As long as it keeps invisible when we do a `kubectl get` afterwards, it should be fine.

Reply:
I made it a pointer. :)